### PR TITLE
Allow deselection of optional SelectField

### PR DESCRIPTION
**Pull request by todda00**
Reference Issue: #119 

Tested with optional, required, with and without placeholders. Also tested with Nested objects and arrays of objects.

If this looks acceptable, I will commit more to this PR/branch for the other 3 UI packages.

### CODE REVIEW

**Comment by radekmie** [comment]
It looks great! I'm waiting for the rest.

**Comment by todda00** [comment]
might want to wait to publish, I have a minor PR I am going to submit for BS4 today.

**Comment by radekmie** [comment]
[There are already few things waiting for release](https://github.com/vazco/uniforms/compare/v1.3.0...master).

File: packages/uniforms-semantic/src/components/fields/SelectField.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ const renderCheckboxes = ({allowedValues, disabled, fieldType, id, name, onChang
     )
 ;
 
-const renderSelect = ({allowedValues, disabled, id, inputRef, name, onChange, placeholder, transform, value}) =>
+const renderSelect = ({allowedValues, disabled, id, inputRef, label, name, onChange, placeholder, required, transform, value}) =>
```
Comment (radekmie) [inline packages/uniforms-semantic/src/components/fields/SelectField.js]: This line is not passing ESLint.

File: packages/uniforms-unstyled/src/components/fields/SelectField.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ const renderCheckboxes = ({allowedValues, disabled, fieldType, id, name, onChang
     )
 ;
 
-const renderSelect = ({allowedValues, disabled, id, inputRef, name, onChange, placeholder, transform, value}) =>
+const renderSelect = ({allowedValues, disabled, id, inputRef, label, name, onChange, placeholder, required, transform, value}) =>
```
Comment (radekmie) [inline packages/uniforms-unstyled/src/components/fields/SelectField.js]: This one too.

File: packages/uniforms-bootstrap3/src/components/fields/SelectField.js [hunk 3]
```diff
@@ -64,4 +64,3 @@ const Select = props =>
 ;
 
 export default connectField(Select);
-
```
Comment (radekmie) [inline packages/uniforms-bootstrap3/src/components/fields/SelectField.js]: It's required by ESLint.

File: packages/uniforms-bootstrap4/src/components/fields/SelectField.js [hunk 4]
```diff
@@ -64,4 +64,3 @@ const Select = props =>
 ;
 
 export default connectField(Select);
-
```
Comment (radekmie) [inline packages/uniforms-bootstrap4/src/components/fields/SelectField.js]: It's required by ESLint.

File: packages/uniforms-semantic/src/components/fields/SelectField.js [hunk 5]
```diff
@@ -33,27 +33,28 @@ const renderCheckboxes = ({allowedValues, disabled, fieldType, id, name, onChang
...
-                {transform ? transform(value) : value}
-            </option>
-        )}
-    </select>
+            {allowedValues.map(value =>
+                <option key={value} value={value}>
+                    {transform ? transform(value) : value}
+                </option>
+            )}
+        </select>
```
Comment (radekmie) [inline packages/uniforms-semantic/src/components/fields/SelectField.js]: Could you please do it exactly the same like in the other places? [Like here](https://github.com/vazco/uniforms/blob/master/packages/uniforms-semantic/src/components/fields/NumField.js).

File: packages/uniforms-unstyled/src/components/fields/SelectField.js [hunk 6]
```diff
@@ -30,27 +30,28 @@ const renderCheckboxes = ({allowedValues, disabled, fieldType, id, name, onChang
...
-                {transform ? transform(value) : value}
-            </option>
-        )}
-    </select>
+            {allowedValues.map(value =>
+                <option key={value} value={value}>
+                    {transform ? transform(value) : value}
+                </option>
+            )}
+        </select>
```
Comment (radekmie) [inline packages/uniforms-unstyled/src/components/fields/SelectField.js]: Same here.

File: packages/uniforms-bootstrap3/src/components/fields/SelectField.js (hunk 3 above)
Comment (radekmie) [inline packages/uniforms-bootstrap3/src/components/fields/SelectField.js]: _(or at least it should)_

Comment (todda00) [inline packages/uniforms-bootstrap3/src/components/fields/SelectField.js]: not sure why the commit removed the line, its showing in my editor, but its not committing it back with the newline at the end, strange.

Comment (todda00) [inline packages/uniforms-bootstrap3/src/components/fields/SelectField.js]: our OS must handle it differently, I am seeing the newline in my editor and the line count on Github indicates the newline is there

**Comment by radekmie** [comment]
Great job.
Thanks, @todda00!

**Pull request merged**